packages/asset/contracts/AssetCreate.sol

```diff
@@ -354,5 +356,267 @@ contract AssetCreate is
+    /// @param metadataHash The metadata hash of the asset
+    /// @param maxSupply The max supply of the asset
+    function _hashLazyMint(
+        address creator,
+        uint16 nonce,
+        uint8 tier,
+        uint256 amount,
+        uint256 unitPrice,
+        address paymentToken,
+        string memory metadataHash,
+        uint256 maxSupply
+    ) internal view returns (bytes32 digest) {
+        digest = _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    LAZY_MINT_TYPEHASH,
+                    creator,
+                    nonce,
+                    tier,
+                    amount,
+                    unitPrice,
+                    paymentToken,
+                    keccak256((abi.encodePacked(metadataHash))),
+                    maxSupply
+                )
+            )
+        );
+    }
+
+    /// @notice Creates a hash of the lazy mint batch data
+    /// @param creators The addresses of the creators
+    /// @param nonce The nonce of the creator
+    /// @param tiers The tiers of the assets
+    /// @param amounts The amounts of copies to mint
+    /// @param unitPrices The unit prices of the assets
+    /// @param paymentTokens The payment tokens of the assets
+    /// @param metadataHashes The metadata hashes of the assets
+    /// @param maxSupplies The max supplies of the assets
+    function _hashLazyBatchMint(
+        address[] memory creators,
+        uint16 nonce,
+        uint8[] memory tiers,
+        uint256[] memory amounts,
+        uint256[] memory unitPrices,
+        address[] memory paymentTokens,
+        string[] memory metadataHashes,
+        uint256[] memory maxSupplies
+    ) internal view returns (bytes32 digest) {
+        digest = _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    LAZY_MINT_BATCH_TYPEHASH,
+                    keccak256(abi.encodePacked(creators)),
+                    nonce,
+                    keccak256(abi.encodePacked(tiers)),
+                    keccak256(abi.encodePacked(amounts)),
+                    keccak256(abi.encodePacked(unitPrices)),
+                    keccak256(abi.encodePacked(paymentTokens)),
+                    _encodeHashes(metadataHashes),
+                    keccak256(abi.encodePacked(maxSupplies))
+                )
+            )
+        );
+    }
+
+    /// @notice Set the lazy mint fee
+    /// @param _lazyMintFee The fee to set
+    function setLazyMintFee(uint256 _lazyMintFee) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        lazyMintFee = _lazyMintFee;
+    }
+
+    /// @notice Set the lazy mint fee receiver
+    /// @param _lazyMintFeeReceiver The receiver to set
+    function setLazyMintFeeReceiver(address _lazyMintFeeReceiver) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(_lazyMintFeeReceiver != address(0), "AssetCreate: Invalid receiver");
+        lazyMintFeeReceiver = _lazyMintFeeReceiver;
+    }
+
     uint256[45] private __gap;
```

Comment:
Are you proposing to leave the gap unchanged @wojciech-turek ?

Reply:
Definitely not! Changed, thanks.